projects/igniteui-angular/src/lib/combo/combo.component.spec.ts

```diff
@@ -1671,24 +397,199 @@ describe('igxCombo', () => {
+            try {
+                combo.data = undefined;
+            } catch (ex) {
+                errorMessage = ex.message;
+            }
+            expect(errorMessage).toBe('');
+            expect(combo.data).not.toBeUndefined();
+            expect(combo.data).not.toBeNull();
+            expect(combo.data.length).toBe(0);
+        });
+        it('should properly handleInputChange', () => {
+            combo = new IgxComboComponent({ nativeElement: null }, mockCdr, mockSelection as any, mockComboService, null, mockInjector);
+            const dropdown = jasmine.createSpyObj('IgxComboDropDownComponent', ['selectItem']);
+            combo.ngOnInit();
+            combo.data = data;
+            combo.dropdown = dropdown;
+            combo.filterable = true;
+            const matchSpy = spyOn<any>(combo, 'checkMatch').and.callThrough();
+            spyOn(combo.onSearchInput, 'emit');
+
+            combo.handleInputChange();
+            expect(matchSpy).toHaveBeenCalledTimes(1);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledTimes(0);
+
+            combo.handleInputChange('Fake');
+            expect(matchSpy).toHaveBeenCalledTimes(2);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledTimes(1);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledWith('Fake');
 
-    describe('Rendering tests: ', () => {
-        it('Should apply all appropriate classes on combo initialization', () => {
-            const fix = TestBed.createComponent(IgxComboScrollTestComponent);
-            fix.detectChanges();
+            combo.handleInputChange('');
+            expect(matchSpy).toHaveBeenCalledTimes(3);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledTimes(2);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledWith('');
 
-            const comboWrapper = fix.nativeElement.querySelector(CSS_CLASS_COMBO);
+            combo.filterable = false;
+            combo.handleInputChange();
+            expect(matchSpy).toHaveBeenCalledTimes(4);
+            expect(combo.onSearchInput.emit).toHaveBeenCalledTimes(2);
+        });
+    });
+    describe('Initialization and rendering tests: ', () => {
+        configureTestSuite();
+        beforeAll(async(() => {
+            TestBed.configureTestingModule({
+                declarations: [
+                    IgxComboSampleComponent
+                ],
+                imports: [
+                    IgxComboModule,
+                    NoopAnimationsModule,
+                    IgxToggleModule,
+                    ReactiveFormsModule,
+                    FormsModule
+                ]
+            }).compileComponents();
+        }));
+        beforeEach(() => {
+            fixture = TestBed.createComponent(IgxComboSampleComponent);
+            fixture.detectChanges();
+            combo = fixture.componentInstance.combo;
+            input = fixture.debugElement.query(By.css(`.${CSS_CLASS_COMBO_INPUTGROUP}`));
```

Comment:
Can you set the constant to include the dot(.) and you will not need to place it in all queries.

Reply:
We do not always need the dot like for example
expect(combo.cssClass).toEqual(CSS_CLASS_COMBO);